lib/src/configuration.dart

```diff
@@ -70,20 +50,119 @@ class Configuration {
     return Directory.current.absolute.path;
   }
 
+  String? get fifoFilesFallbackPath;
+  String get path;
+  RealmSchema get schema;
+  int get schemaVersion;
+  List<int>? get encryptionKey;
+
+  @Deprecated('Use Configuration.local instead')
+  factory Configuration(
+    List<SchemaObject> schemaObjects, {
+    Function(Realm realm)? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+    bool disableFormatUpgrade,
+    bool isReadOnly,
+    bool Function(int totalSize, int usedSize)? shouldCompactCallback,
+  }) = LocalConfiguration;
+
+  factory Configuration.local(
+    List<SchemaObject> schemaObjects, {
+    Function(Realm realm)? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+    bool disableFormatUpgrade,
+    bool isReadOnly,
+    bool Function(int totalSize, int usedSize)? shouldCompactCallback,
+  }) = LocalConfiguration;
+
+  factory Configuration.inMemory(
+    List<SchemaObject> schemaObjects,
+    String identifier, {
+    Function(Realm realm)? initialDataCallback,
+    int schemaVersion,
```

Comment:
We do in dotnet, but you are right, I don't really see the point either.

Reply:
Yeah, there may be some indirectness in the configs in .NET that we haven't fixed because it's low priority. We should definitely review them at some point.

Reply:
Actually you cannot open an in-memory realm on core without setting the version, but we can set it to 0 always.